src/theme/default/components/gridblock/components/variants/DynamicComponentRenderer.tsx

```diff
@@ -0,0 +1,70 @@
+"use client";
+import React from "react";
+import { PageVariant } from "@conversiondigital/headless-basics-data/src/cms/constants";
+import { getLogger } from "@conversiondigital/headless-basics-data/src";
+import { 
+  ToggleComponent, 
+  TextBlockComponent, 
+  HeroComponent, 
+  MottoComponent 
+} from '../subcomponents';
+
+interface DynamicComponentRendererProps {
+  item: any;
+  loadedComponent: any;
+  blueprint: any;
+}
+
+const log = getLogger("default.components.sanity.gridblock.dynamicRenderer");
+
+export default function DynamicComponentRenderer({ 
+  item, 
+  loadedComponent, 
+  blueprint 
+}: DynamicComponentRendererProps) {
+  log.debug(`Rendering component for type: ${item._type} with data:`, loadedComponent?.data);
+
+  const componentProps = {
+    matchingData: loadedComponent.data,
+    componentInformation: {
+      uid: item._key,
+      identifier: item._key,
+      pageDefinition: {
+        source: item._type,
+        pageIdentifier: {
+          identifier: item._key,
+          pageVariant: "subComponentsPage" as PageVariant,
+          isFixedLayout: false
+        }
+      }
+    },
+    blueprint,
+    item
+  };
+
+  switch (item._type.toLowerCase()) {
+    case 'textblock':
```

Comment:
These subcomponent lookups will be done in the components own query file.  Via loadSingleComponentGraphQLData